src/quote/Quote.ts

```diff
@@ -0,0 +1,354 @@
+    }
+
+    if (typeof slippage !== 'number' || slippage < 0 || slippage > 1) {
+      return {
+        result: undefined,
+        error: { message: 'Invalid slippage: must be a number between 0 and 1' },
+      };
+    }
+
+    // Handle ipfsCidOrContent validation (either a string CID or a content object)
+    const isString = typeof ipfsCidOrContent === 'string';
+    const isObject = typeof ipfsCidOrContent === 'object' && ipfsCidOrContent !== null;
+
+    if (!isString && !isObject) {
+      return {
+        result: undefined,
+        error: { message: 'Invalid ipfsCidOrContent: must be a string CID or content object' },
+      };
+    }
+
+    if (isString && ipfsCidOrContent !== '') {
+      const isValidCID = this.ipfs.validateIPFSCid(ipfsCidOrContent as string);
+      if (!isValidCID) {
+        return {
+          result: undefined,
+          error: { message: 'Invalid ipfsCid: must be a valid IPFS CID' },
+        };
+      }
+    }
+
+    const productType = productsMap[productId]?.productType;
+    if (productType === undefined) {
+      return {
+        result: undefined,
+        error: { message: `Invalid product` },
+      };
+    }
+
+    if (IPFS_CONTENT_TYPE_BY_PRODUCT_TYPE[productType] !== undefined && !ipfsCidOrContent) {
+      return {
+        result: undefined,
+        error: {
+          message: `Missing IPFS content. \n
+          ${productTypes[productType]?.name} requires ${IPFS_CONTENT_TYPE_BY_PRODUCT_TYPE[productType]} content type.`,
+        },
+      };
+    }
+
+    let ipfsData = ipfsCidOrContent as string;
+    const contentType = IPFS_CONTENT_TYPE_BY_PRODUCT_TYPE[productType];
+
+    // Handle uploading content to IPFS if provided as an object
+    if (!isString && contentType !== undefined && ipfsCidOrContent) {
+      try {
+        // Use the IPFS instance to upload content
+        ipfsData = await this.ipfs.uploadIPFSContent([contentType, ipfsCidOrContent] as IPFSTypeContentTuple);
+      } catch (error: unknown) {
+        return {
+          result: undefined,
+          error: { message: (error as Error).message || 'Failed to upload IPFS content' },
+        };
+      }
+    }
+
+    // Convert slippage from 0-1 to 0-100_00
+    const slippageValue = slippage * SLIPPAGE_DENOMINATOR;
+    const quoteParams: QuoteParams = {
+      productId,
+      amount,
+      period,
+      coverAsset: coverAssetEnum,
+      paymentAsset: paymentAssetEnum,
+    };
+
+    if (coverId) {
+      quoteParams.coverEditId = coverId;
+    }
+
+    try {
+      // Get quote using helper method
+      const { quote } = await this.getQuote(quoteParams);
+
+      const maxPremiumInAsset = this.calculatePremiumWithCommissionAndSlippage(
+        BigInt(quote.premiumInAsset),
+        DEFAULT_COMMISSION_RATIO,
+        slippageValue,
+      );
+      const yearlyCostPerc = this.calculatePremiumWithCommissionAndSlippage(
+        BigInt(quote.annualPrice),
+        DEFAULT_COMMISSION_RATIO,
```

Comment:
And here `commissionRatio || BUY_COVER_COMMISSION_RATIO_BY_PRODUCT_TYPE[productType],`